guides/common/modules/proc_configuring-pulpcore-api-workers.adoc

```diff
@@ -0,0 +1,31 @@
+[id="Configuring_Pulpcore_API-Workers_{context}"]
+= Configuring Pulpcore API Workers
+
+The pulpcore-api workers responds to incoming API requests. Less api workers consume less memory and results in better performance.
+
+For example, we have compared these two setups:
+
+[width="100%",cols="50%,50%",options="header",]
+|===
+|{Project} VM with 8 CPUs, 32 GiB RAM, 3 pulpcore-api_workers |{Project} VM with 8 CPUs, 32 GiB RAM, 10 pulpcore-api_workers
+|===
+
+Outcome looks similar in both cases, so the recommendation would be to lower the number of api workers  because of the improvement in memory consumption.
+
+.Use the below procedure to tune the pulpcore API workers:
+
+. Update the pulpcore api_service_worker_count on your {ProjectServer} or {SmartProxyServer} in the custom-hiera.yaml file.:
```

Comment:
```suggestion
. Update the pulpcore api_service_worker_count on your {ProjectServer} or {SmartProxyServer} in the `custom-hiera.yaml` file.:
```

I am unsure if we should recommend users editing this file vs. using foreman-installer.

Reply:
I've always taken a stance that `custom-hiera.yaml` is unsupported. Anything that's written in docs, so we can't refer to it.

If there's always a recommendation to lower the number of workers, that should be filed as a bug. Telling users "we have poor defaults, always change this" is a poor experience.

My suggestion is to drop this chapter altogether.

Reply:
And if it is common to change it, an RFE to add a parameter as was done with the worker count is also a valid solution.

Reply:
Hello @ekohl: I totally agree on `custom-hiera.yaml is unsupported` part. Do you mean to dropping off this section or proc_configuring-pulpcore-workers as well?

Reply:
I'm questioning the usefulness of `proc_configuring-pulpcore-workers`. When I read it, it comes down to "here's a parameter you can change, but we noticed it doesn't make a difference". So as a reader of that, is it adding any value or just confusing me?

Reply:
cool, let's drop this section.

Reply:
> Regarding `proc_configuring-pulpcore-api-workers.adoc` section, can we suggest updating pulpcore api_service_worker_count with another method, for example - via foreman-installer?

Currently this is the code that sets it:
https://github.com/theforeman/puppet-pulpcore/blob/297a48acafcbdb9871689b6e267598c7414692c2/manifests/init.pp#L246-L247

https://projects.theforeman.org/issues/36957 was opened by @maximiliankolb to expose a parameter for this and I just opened https://github.com/theforeman/puppet-foreman_proxy_content/pull/467.

I still think that most users shouldn't touch this and if our default tuning is poor, then it should be fixed.